### ISSUE TITLE
Unable to destroy activity null pointer exception at barcodecapture.ondestroy

### ISSUE THREAD

**Issue by Ssehgal29** [KingsMentor/MobileVisionBarcodeScanner#69]
whenever i call  onbackpressed or perform an intent the app crashes and show the error unable to destroy the activity Attempt to invoke virtual method 'java.lang.Thread$State java.lang.Thread.getState()' on a null object reference.

**Comment by damms005** [comment]
@Ssehgal29 were you able to get around this??

**Comment by Ssehgal29** [comment]
Yess it's because of gradle incompatibility... Just go to project structure and set your gradle to 4.0.0 and everything works perfectly.

**Comment by KushalMehta1995** [comment]
@damms005 were you able to get around this??

**Comment by KushalMehta1995** [comment]
@Ssehgal29 were you able to get around this?? It's not working in my case.

**Comment by damms005** [comment]
@Kush1195 Yes. I had to clone this repo locally and rebuild the app locally, and included it in my app as a local library dependency (from my project's graddle settings)

**Comment by KushalMehta1995** [comment]
> @Kush1195 Yes. I had to clone this repo locally and rebuild the app locally, and included it in my app as a local library dependency (from my project's graddle settings)

Please Do soon. Thank You

**Comment by kprathap23** [comment]
Any update on this issue. How can we fix this.. i tried to change gradle to 4.0.0  still not working. please let me know if any one fixed and the related all gradle configurations . I used this library in a flutter plugin https://pub.dev/packages/mvbarcodescan. Now i am getting this crash issue

**Comment by Fqceless** [comment]
To anyone still having this problem, hopefully this helps.
I was trying to get around this for hours, and the solution I found was this:
```
protected void onDestroy(){
        try {
            super.onDestroy();
        } catch (Exception e) {
            super.finishAndRemoveTask();
        }
    }
}
```

The class I was using would throw this error every time it got destroyed (super.onDestroy() was called), and it had something to do with the onDestroy methods being inherited by this repo.
I was able to just catch the exception and run this random function to finish the task after the error was thrown.
So far after my testing, this solution has held up.

Hopefully this can help, or give you direction to fix it!

Note: This worked for Gradle version 8.0.2

**Comment by abhishekdeswal03** [comment]
8.0.2 give error can't find such lib "xyz.belvi.mobilevision:barcodescanner:2.0.3"

@Fqceless
